src/reporting/JSON_report.ml

```diff
@@ -211,10 +211,15 @@ let rec taint_call_trace = function
       let* call_trace = taint_call_trace call_trace in
       Some (Out.CoreCall (location, intermediate_vars, call_trace))
 
-let taint_trace_to_dataflow_trace { source; tokens; sink } :
+let taint_trace_to_dataflow_trace { sources; sink } :
     Out.core_match_dataflow_trace =
+  let call_trace, tokens =
+    match sources with
+    | [] -> raise Common.Impossible
+    | (call_trace, tokens) :: _ -> (call_trace, tokens)
```

Comment:
You could add a comment here to explain why we are ignoring all the other taints. This may be confusing for others without some clarification.